<h1> LINEAR MODELING USING THE F DISTRIBUTION

>Nguyen Tran

George Washington University

Introduction to Health Data Science PUBH1142

April 14, 2026

### Import Packages

In [1]:
import numpy as np
from scipy import stats
from scipy import special
from pandas import DataFrame
import plotly.graph_objects as go
import plotly.express as px
import plotly.figure_factory as ff
import plotly.io as pio
pio.templates.default = 'plotly_white'
from patsy import dmatrices
import statsmodels.api as sm

In [4]:
%config InlineBackend.figure_format = 'retina'

## Quiz Questions

In [5]:
# Seed the pseudo-random number generator
np.random.seed(12)

# Sample size
n=100

# Create data
wcc = np.round(np.random.normal(15, 5, n), 0)
crp = (wcc * 2) + np.round(np.random.normal(0, 10, n), 0)

lipase = wcc + crp + np.round(np.random.normal(2, 10, n), 0)

# Pandas dataframe obejct
df = DataFrame(
    {'WCC':wcc,
     'CRP':crp,
     'Lipase':lipase}
)

1. ##### *Create multiple linear regression model with `WCC` and `CRP` as predictors of `Lipase`.*


In [6]:
# Create multiple linear regression model with `WCC` and `CRP` as predictors of `Lipase`
y, X = dmatrices('Lipase ~ WCC + CRP', data=df, return_type='dataframe')
model = sm.OLS(y, X)
results = model.fit()
print(results.summary())

                            OLS Regression Results                            
Dep. Variable:                 Lipase   R-squared:                       0.764
Model:                            OLS   Adj. R-squared:                  0.759
Method:                 Least Squares   F-statistic:                     157.3
Date:                Tue, 14 Apr 2026   Prob (F-statistic):           3.61e-31
Time:                        09:14:03   Log-Likelihood:                -374.15
No. Observations:                 100   AIC:                             754.3
Df Residuals:                      97   BIC:                             762.1
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      2.3741      3.013      0.788      0.4

2. ##### *Comment on the $F$ statistic.*

The $F$-statistic is large, meaning the results can vary by up to 157.3 points.

3. ##### *Comment on the individual coefficients and their $p$ values.*

The coefficient for `WCC`is positive but small, meaning the correlation is weak. THe coefficient for `CRP` is also positive and above `1`, so the correlation is very strong. The P values for both `WCC` and `CRP` are below $0.05$, so the recorded data is likely not due to chance. The $p$-value for `CRP` beinge extremely statistically significant.

4. Comment on the $R^2$ value.

The $R$-squared value is adjusted to 0.759 to take into account the multiple dependent variables used in the linear regression model. The adjusted $R$-squared value shows that $75.9$% of the variance of the dependent variable is explained by the independent variables in the model.